File: w07/d01/Sophie_Laffont/hangman-hw-week7-day01/hangman.js
function Hangman(word){

var playerProgress = Array(word.length+1).join("_").split("");
var position;
var guesses=0;

  this.progress = function(){
    // for (var i =0; i < word.length; i++){
    //   playerProgress.push("_");
    // } //  string => array

  //playerProgress = playerProgress.join(""); // array => string
  return playerProgress.join(""); // array => string
  };

this.guess= function(letter) {
  var array = word.split(""); // word becomes array of letters
  var correctGuess = false;
  for (var i = 0; i <array.length; i++) {
    if (array[i] === letter) { // check to see if letter is the array
       playerProgress[i] = letter; // put the letter at the same index position in player array
      correctGuess = true;
    };
  };
  if (!correctGuess){
    guesses+=1;
    if (guesses >= 5) {
      return null;
    }
    return "Nope. Try again";
  }
  else if(guesses >= 5) {
    return null;
  }
  else if (playerProgress.indexOf("_") == -1 ) {
    return "You won! The word was: "+ playerProgress.join("");
  }
  else if (correctGuess) {
    return playerProgress.join("");
  }

}

this.guesses = function() {
  return guesses;
}
}

//CALLING FUNCTIONS
//var game = new Hangman("hello");

File: w10/d01/Liz_Goldstein/hangman/lib/hangman.rb
class Hangman

  def initialize(word)
    @orginal = word
    @word = word.chars
    @progress = @word.map { "_" }
    @counter = 0
  end

  def progress()
    @progress.join(" ")
  end

  def guess?(guessed_letter)
  if @counter < 5
    if @word.include?(guessed_letter)
      @word.each_with_index do |letter, index|
        if letter == guessed_letter
          @progress[index] = letter
        end
      end
    if @orginal  == @progress.to_s.gsub(" ","")
      return "Won!!"
    else
     true
    end
  else
    @counter +=1
    return "#{@counter} guesses made"
  end
else
  return "GAME OVER"
end
end

  def game_counter
    if @counter < 5
      return @counter
    else
      return "game over"
    end
  end

def giveUp
  @counter = 5
  return @orginal
end

end

